### PR TITLE
Chore: upgrade eslint

### PR DESCRIPTION
**Pull request by philipyoo**
List of Eslint rules: https://eslint.org/docs/rules/

Note: Will squash commits after making any changes based on peoples' feedback

### CODE REVIEW

File: examples/javascript-V4Auth-example.js [hunk 1]
```diff
@@ -1,5 +1,5 @@
 const http = require('http');
-const aws4 = require('aws4');
+const aws4 = require('aws4');  // eslint-disable-line
```
Comment (bennettbuchanan) [inline examples/javascript-V4Auth-example.js]: Seems like maybe we should just disable linting of the examples directory if that's not difficult to do.

Comment (philipyoo) [reply examples/javascript-V4Auth-example.js]: I like the idea :) Added changes to ignore the `examples` dir

Comment (rahulreddy) [reply examples/javascript-V4Auth-example.js]: Seems to be a linter error
```
/home/ubuntu/utapi/examples/javascript-V4Auth-example.js
  2:22  error  Unable to resolve path to module 'aws4'  import/no-unresolved

```

**Comment by rahulreddy** [comment]
@ironman-machine r+

**Comment by philipyoo** [comment]
@ironman-machine try

**Comment by rahulreddy** [comment]
@ironman-machine r- r+

**Pull request merged**